src/Type/Object/FriendshipType.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * Register the Friendship object type and queries.
+ *
+ * @package \WPGraphQL\Extensions\BuddyPress\Type\Object
+ * @since 0.0.1-alpha
+ */
+
+namespace WPGraphQL\Extensions\BuddyPress\Type\Object;
+
+use GraphQL\Error\UserError;
+use GraphQLRelay\Relay;
+use WPGraphQL\AppContext;
+use WPGraphQL\Data\DataSource;
+use WPGraphQL\Extensions\BuddyPress\Data\Factory;
+use WPGraphQL\Extensions\BuddyPress\Model\Friendship;
+
+/**
+ * FriendshipType Class.
+ */
+class FriendshipType implements RegisterInterface {
+
+	/**
+	 * Name of the type.
+	 *
+	 * @var string Type name.
+	 */
+	public static $type_name = 'Friendship';
+
+	/**
+	 * Register the friendship type and queries to the WPGraphQL schema.
+	 */
+	public static function register() {
+		register_graphql_object_type(
+			self::$type_name,
+			[
+				'description'       => __( 'Info about a BuddyPress Friendship.', 'wp-graphql-buddypress' ),
+				'fields'            => [
+					'id' => [
+						'type'        => [ 'non_null' => 'ID' ],
+						'description' => __( 'The globally unique identifier for the friendship.', 'wp-graphql-buddypress' ),
+					],
+					'friendshipId' => [
+						'type'        => 'Int',
+						'description' => __( 'The id field that matches the BP_Friends_Friendship->id field.', 'wp-graphql-buddypress' ),
+					],
+					'initiator' => [
+						'type'        => 'User',
+						'description' => __( 'The initiator of the friendship.', 'wp-graphql-buddypress' ),
+						'resolve'     => function( Friendship $friendship, array $args, AppContext $context ) {
+							return ! empty( $friendship->initiator )
+								? DataSource::resolve_user( $friendship->initiator, $context )
+								: null;
+						},
+					],
+					'friend' => [
+						'type'        => 'User',
+						'description' => __( 'The friend, the one invited to the friendship.', 'wp-graphql-buddypress' ),
+						'resolve'     => function( Friendship $friendship, array $args, AppContext $context ) {
+							return ! empty( $friendship->friend )
```

Comment:
There is a bug here that is returning no information because the WPGraphQL plugin `User` model is validating the user and marking it as private.

Reply:
Fixed!